sale_order_add_variants/wizard/sale_add_variants.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Hugo Santos
+#    Copyright 2015 FactorLibre
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import api, models, fields, _
+
+
+class SaleAddVariants(models.TransientModel):
+    _name = 'sale.add.variants'
+
+    product_tmpl_id = fields.Many2one('product.template', string="Template",
+                                      required=True)
+    variant_line_ids = fields.One2many('sale.add.variants.line', 'wizard_id',
+                                       string="Variant Lines")
+
+    @api.multi
+    def clear_previous_selections(self):
+        for line in self.variant_line_ids:
```

Comment:
this will fail if self has more than one record.

you should rewrite this as:

    self.mapped('variant_line_ids').unlink()